shared/desktop/app/installer.js

```diff
@@ -34,15 +42,109 @@ export default (callback: (err: any) => void): void => {
   if (app.getLoginItemSettings().wasOpenedAtLogin) {
     timeout = 90
   }
-  const args = ['--debug', '--app-path=' + bundlePath, '--run-mode=' + runMode, '--timeout=' + timeout]
+  const args = ['--debug', 'install-auto', '--format=json', '--timeout=' + timeout + 's']
 
-  exec(installerPath, args, 'darwin', 'prod', true, function(err) {
-    if (err && err.code === 1) {
-      // The installer app returns exit status 1, if there was an error and
-      // the user chooses to quit the app.
-      quit()
+  exec(keybaseBin, args, 'darwin', 'prod', true, (err, attempted, stdout, stderr) => {
+    let installResult: InstallResult = {errors: [], hasCLIError: false}
+    if (err) {
+      installResult.errors = [`There was an error trying to run the install (${err.code}).`]
+    } else if (stdout !== '') {
+      try {
+        const result = JSON.parse(stdout)
+        if (result) {
+          installResult = checkErrors(result)
+        } else {
+          installResult.errors = [`There was an error trying to run the install. No output.`]
+        }
+      } catch (err) {
+        installResult.errors = [
+          `There was an error trying to run the install. We were unable to parse the output of keybase install-auto.`,
+        ]
+      }
+    }
+
+    if (installResult.errors.length > 0) {
+      showError(installResult.errors, callback)
+      return
+    }
+
+    // If we had an error install CLI, let's prompt and try to do it via
+    // privileged install.
+    if (installResult.hasCLIError && !installerState.state.promptedForCLI) {
+      promptForInstallCLIPrivileged(keybaseBin, callback)
       return
     }
-    callback(err)
+
+    callback(null)
   })
 }
+
+function checkErrors(result: any): InstallResult {
```

Comment:
Do we know what Type `result` is from the flow-types? Because if we did, it would be super useful to have this types. Seems kinda brittle